pay-api/src/pay_api/services/refund.py

```diff
@@ -268,12 +271,43 @@ def create_refund(cls, invoice_id: int, request: Dict[str, str], **kwargs) -> Di
             payment_method=invoice.payment_method_code
         )
         payment_account = PaymentAccount.find_by_id(invoice.payment_account_id)
-        invoice_status = pay_system_service.process_cfs_refund(invoice, payment_account=payment_account)
+        refund_partial_lines = cls._get_partial_refund_lines(request.get('refundRevenue', None))
+        invoice_status = pay_system_service.process_cfs_refund(invoice=invoice,
+                                                               payment_account=payment_account)
         refund.flush()
+        cls._save_partial_refund_lines(refund_partial_lines)
         message = REFUND_SUCCESS_MESSAGES.get(f'{invoice.payment_method_code}.{invoice.invoice_status_code}')
         # set invoice status
         invoice.invoice_status_code = invoice_status or InvoiceStatus.REFUND_REQUESTED.value
         invoice.refund = invoice.total  # no partial refund
         invoice.save()
         current_app.logger.debug(f'Completed refund : {invoice_id}')
         return {'message': message}
+
+    @staticmethod
+    def _save_partial_refund_lines(partial_refund_lines: List[RefundPartialLine]):
+        """Persist a list of partial refund lines."""
+        for line in partial_refund_lines:
+            refund_line = RefundPartialModel(
+                payment_line_item_id=line.line_item_id,
+                refund_amount=line.refund_amount,
+                refund_type=line.refund_type
+            )
+            db.session.add(refund_line)
+
+    @staticmethod
+    def _get_partial_refund_lines(refund_revenue: List[Dict]) -> List[RefundPartialLine]:
+        """Convert Refund revenue data to a list of Partial Refund lines."""
+        if not refund_revenue:
+            return []
+
+        refund_lines = []
+        for refund in refund_revenue:
```

Comment:
Investigating an issue when using the converter where the payload is camel case but the data class is snake case.

Trying to update the converter to allow for a parameter to do a camel to snake case conversion when it is instantiated.

Reply:
Updated converter to handle camel to snake conversion